python/tskit/tables.py

```diff
@@ -112,6 +112,21 @@ class MutationTableRow:
     metadata: bytes
     time: float
 
+    def __eq__(self, other):
+        return (
```

Comment:
Looks like you need an ``isinstance(other, MutationTableRow)`` here and return False otherwise.